platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/ClientInfoFilter.java

```diff
@@ -40,48 +41,47 @@
  * <p>For example, the key associated with {@link ServletRequest#getRemoteAddr()} would be the
  * string {@code remoteAddr}.
  *
- * <p>The only exception to this rule, {@link ClientInfoFilter#CLIENT_INFO_KEY}, which holds a value
- * string of {@code client-info}, is the key used to access the entire client information map. It
- * may contain different kinds of data that does not necessarily correlate to the servlet API.
+ * <p>The only exception to this rule, {@link ThreadContextProperties#CLIENT_INFO_KEY}, which holds
+ * a value string of {@code client-info}, is the key used to access the entire client information
+ * map. It may contain different kinds of data that does not necessarily correlate to the servlet
+ * API.
  */
-public class ClientInfoFilter implements HttpFilter {
-
+public class ClientInfoFilter implements HttpFilter, SecurityFilter {
```

Comment:
Updated this class to implement SecurityFilter as well as HttpFilter.  This ensures that this executes during the security filter chain as well as the normal HTTP filter chain.  Without this, audit events that occur during security filter chain would be missing the client IP/port.

Reply:
I was trying to think of a way we might be able to achieve this such that the thread context isn't clobbered in the first place. Here's a thought:

Shiro allows binding the subject to the thread directly (https://shiro.apache.org/static/1.3.2/apidocs/org/apache/shiro/util/ThreadContext.html#bind-org.apache.shiro.subject.Subject-) rather than using Subject::execute(). That would preserve any parameters set here. We'd just have to make sure to unbind it ourselves then at the end.
```
ThreadContext.bind(subject)
try {
   // Do stuff... 
}
finally {
  ThreadContext.unbindSubject()
}
```

That would work if we're not creating a new thread. If we are (like in the SecureWebSocketServlet), we could save off the existing thread context first, then restore it afterwords to make sure that nothing is lost (see https://shiro.apache.org/static/1.3.2/apidocs/org/apache/shiro/util/ThreadContext.html#getResources--):
```
ExecutorService executor = Executors.newFixedThreadPool(2)
Map<Object, Object> threadLocalMap = ThreadContext.getResources()
executor.submit(() -> {
    ThreadContext.setResources(threadLocalMap)
    subject.execute(runnable);
});
```

@stustison is there a reason why didn't do it this way originally?

Reply:
Yeah it would bind it to the current thread and then the next connection going through will flip it to a different subject underneath the first request. As soon as you have a decent amount of traffic it will break.

Reply:
Subject::execute() doesn't create a new thread, though, so if that's true I'm struggling to understand why it doesn't suffer the same problems https://github.com/apache/shiro/blob/main/core/src/main/java/org/apache/shiro/subject/Subject.java#L446-L457

Reply:
Don't mean to block this PR though given that it works and has been tested. Just would be curious if that doesn't work why not